compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -744,6 +759,12 @@ private void validateWorkerAnnAttachments(BLangExpression expr) {
 
     public void visit(BLangRecordVariable varNode) {
 
+        // Only simple variables are allowed to be configurable.
+        if (isConfigurable(varNode)) {
+            dlog.error(varNode.pos, DiagnosticCode.ONLY_SIMPLE_VARIABLES_ARE_ALLOWED_TO_BE_CONFIGURABLE);
+            return;
```

Comment:
Same as the previous comment. Log the error and continue.

Reply:
Handle this in other places as well.

Reply:
Done